packages/devtools_app/lib/src/memory/memory_android_chart.dart

```diff
@@ -0,0 +1,324 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_shared/devtools_shared.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../auto_dispose_mixin.dart';
+import '../charts/chart.dart';
+import '../charts/chart_controller.dart';
+import '../charts/chart_trace.dart' as trace;
+import '../theme.dart';
+
+import 'memory_controller.dart';
+import 'memory_timeline.dart';
+
+class MemoryAndroidChart extends StatefulWidget {
+  const MemoryAndroidChart(this.chartController);
+
+  final ChartController chartController;
+
+  @override
+  MemoryAndroidChartState createState() => MemoryAndroidChartState();
+}
+
+/// Name of each trace being charted, index order is the trace index
+/// too (order of trace creation top-down order).
+enum TraceName {
+  stack,
+  graphics,
+  nativeHeap,
+  javaHeap,
+  code,
+  other,
+  system,
+  total,
+}
+
+class MemoryAndroidChartState extends State<MemoryAndroidChart>
+    with AutoDisposeMixin {
+  /// Controller attached to the chart.
+  ChartController get _chartController => widget.chartController;
+
+  /// Controller for managing memory collection.
+  MemoryController _memoryController;
+
+  MemoryTimeline get _memoryTimeline => _memoryController.memoryTimeline;
+
+  ColorScheme colorScheme;
+
+  @override
+  void initState() {
+    setupTraces();
+
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    _memoryController = Provider.of<MemoryController>(context);
+
+    // TODO(jacobr): this is an ugly way to be using the theme. It would be
+    // better if the controllers weren't involved with the color scheme.
+    colorScheme = Theme.of(context).colorScheme;
+
+    //_initController(colorScheme);
+
+    cancel();
+
+    setupTraces();
+    setupChartData();
+
+    addAutoDisposeListener(_memoryTimeline.sampleAddedNotifier, () {
+      _processHeapSample(_memoryTimeline.sampleAddedNotifier.value);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_chartController != null) {
+/*
```

Comment:
remove commented out code. Bonus points for removing the cached color scheme everywhere as that shouldn't be needed now that your chart code is more clearly structured like a typical flutter widget.

Reply:
Done.